wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/convert/DefaultEditorProvisioner.java

```diff
@@ -110,9 +112,32 @@ private void provisionDefaultPlugins(
       }
     }
 
-    missingPluginsIdToRef
-        .values()
-        .forEach(pluginRef -> components.add(new ComponentImpl(PLUGIN_COMPONENT_TYPE, pluginRef)));
+    for (String pluginRef : missingPluginsIdToRef.values()) {
+      ExtendedPluginFQN fqn;
+      ComponentImpl component;
+      try {
+        fqn = pluginFQNParser.parsePluginFQN(pluginRef);
+      } catch (InfrastructureException e) {
+        throw new DevfileException(e.getMessage(), e);
+      }
+      if (isNullOrEmpty(fqn.getReference())) {
+        component = new ComponentImpl(PLUGIN_COMPONENT_TYPE, pluginRef);
+      } else {
+        try {
+          fqn = pluginFQNParser.evaluateFqn(pluginRef, contentProvider);
+          component = new ComponentImpl(PLUGIN_COMPONENT_TYPE, pluginRef);
+        } catch (InfrastructureException e) {
+          throw new DevfileException(e.getMessage(), e);
+        }
+        if (!isNullOrEmpty(fqn.getId())) {
+          component.setId(fqn.getId());
+        }
+        if (!isNullOrEmpty(fqn.getReference())) {
+          component.setReference(fqn.getReference());
+        }
+      }
+      components.add(component);
+    }
```

Comment:
Will it be possible to extract this in a separate method(s) + provide test coverage for this particular method + doc.
The logic is not straight-forward for me atm